File: Python/Algorithms/Sorting/Comb_sort.py
#comb sort (variation of bubble sort)
def comb_sort(inputArray):
    swap = True
    gap = len(inputArray)
    while gap >1 or swap:
        gap = max(1, int(gap/1.3))
        swap = False
        for i in range(len(inputArray) - gap):
            if inputArray[i] > inputArray[i+gap]:
                inputArray[i],inputArray[i+gap] = inputArray[i+gap], inputArray[i]
                swap = True

inputArray = input('Enter the list space seperated that needs to be sorted: ')
inputArray = [int(i) for i in inputArray.split()]
comb_sort(inputArray)
print('Final sorted list is:',inputArray)

File: Ruby/Algorithms/Sorting/CombSort.rb
# Comb sort: A variation on bubble sort that dramatically improves performance.
# Source: http://yagni.com/combsort/
# Requirements: Needs to be able to compare elements with <=>, and the [] []= methods should
# be implemented for the container.
# Time Complexity: О(n^2)
# Space Complexity: О(n) total, O(1) auxiliary
# Stable: Yes
# 
# comb_sort [5, 4, 3, 1, 2] => [1, 2, 3, 4, 5]
def comb_sort(container)
    container
    gap = container.size
    loop do
        gap = gap * 10/13
        gap = 11 if gap == 9 || gap == 10
        gap = 1 if gap < 1
        swapped = false
        (container.size - gap).times do |i|
        if (container[i] <=> container[i + gap]) == 1
            container[i], container[i+gap] = container[i+gap], container[i] # Swap
            swapped = true
        end
        end
        break if !swapped && gap == 1
    end
    container
end